mm2src/mm2_net/src/event_streaming/wasm_event_stream.rs

```diff
@@ -11,21 +11,9 @@ struct SendableMessagePort(web_sys::MessagePort);
 unsafe impl Send for SendableMessagePort {}
 
 /// Handles broadcasted messages from `mm2_event_stream` continuously for WASM.
-pub async fn handle_worker_stream(ctx: MmArc) {
-    let config = ctx
-        .event_stream_configuration
-        .as_ref()
-        .expect("Event stream configuration couldn't be found. This should never happen.");
-
-    let mut channel_controller = ctx.stream_channel_controller.clone();
-    let mut rx = channel_controller.create_channel(config.total_active_events());
-
-    let worker_path = config
-        .worker_path
-        .to_str()
-        .expect("worker_path contains invalid UTF-8 characters");
+pub async fn handle_worker_stream(ctx: MmArc, worker_path: String) {
     let worker = SendableSharedWorker(
-        SharedWorker::new(worker_path).unwrap_or_else(|_| {
+        SharedWorker::new(&worker_path).unwrap_or_else(|_| {
             panic!(
```

Comment:
Should we panic if browser is not supported?

Reply:
or we could just log the error and return. this is a spawned task and won't affect the rest of the runtime.

Reply:
Maybe the js caller may hang or crash